### PR TITLE
Make private key and known_hosts cluster specific

### PR DESCRIPTION
**Pull request by mkjpryor-stfc**

### CODE REVIEW

File: tasks/main.yml [hunk 1]
```diff
@@ -36,9 +47,10 @@
         - cluster_gw_group is defined
         - cluster_gw_user is not defined or cluster_gw_ip is not defined
 
-    - name: Set private key file fact
+    - name: Set facts for private key and known_hosts files
       set_fact:
-        cluster_private_key_file: "{{ private_data_dir }}/{{ cluster_name }}_ssh_private_key"
+        cluster_private_key_file: "{{ private_data_dir }}/{{ openstack_project_id }}_{{ cluster_name }}_ssh_private_key"
```
Comment (brtkwr) [inline tasks/main.yml]: maybe call this `ssh_private_key.{{ openstack_project_id }}.{{ cluster_name }}`

Comment (brtkwr) [reply tasks/main.yml]: wait, isn't it enough to have a known_hosts file per project? Why cluster level isolation? Its going to get littered with lots of files. Its unlikely you'll have a user creating multiple cluster per project at the same time, and even then, it is robust enough at that level.

Comment (mkjpryor-stfc) [reply tasks/main.yml]: We need an SSH private key file per cluster (which means it needs project ID and cluster name in the filename).

The renaming might be better though.

The known_hosts file can technically be per-project, but I quite like it per cluster as well.

**Review by brtkwr — APPROVED** [review]

**Comment by brtkwr** [comment]
Approving as I am keen to test it! Dont want to create a commit repointing jasmin appliances to this branch :)

**Pull request merged**